hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -824,6 +824,22 @@ private JobAndSqlSummary getJobAndSqlSummary(LightMasterContext lmc) {
                 false);
     }
 
+    /**
+     * Sets the specified configuration if the given job is suspended.
+     * Otherwise, an {@link IllegalStateException} is thrown by the returned future.
+     */
+    public CompletableFuture<Void> setJobConfig(long jobId, @Nonnull JobConfig config) {
+        return runWithJob(jobId,
+                masterContext -> masterContext.setJobConfig(config),
```

Comment:
I think that job config update should be guarded by `mc.lock()`. `runWithJob` without locks is used for read operations (with some complicated measures to return sensible data) and does not apply the lock itself. `setJobConfig()` could execute in parallel with for example `resumeJob()`